src/extensionVariables.ts

```diff
@@ -0,0 +1,13 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { IAzureNode } from "vscode-azureextensionui";
+
+/**
+ * Namespace for common variables used throughout the extension. They must be initialized in the activate() method of extension.ts
+ */
+export namespace ext {
+    export let connectedMongoDB: IAzureNode | undefined;
```

Comment:
Do you actually need to keep a reference to the node instead of just the ID?  Would keeping only the ID break your dependency and allow you to reduce the complexity?

Reply:
I need the node to fix your other comment :) (making sure I refresh the "old" node in case you switch from Azure to Attached)

Reply:
Can you search for the node?

Reply:
Doesn't matter much if this doesn't reduce complexity.

Reply:
Yes, but that feels more complex and slower than just keeping a reference to the node IMO